bundles/org.openhab.ui/web/src/pages/settings/items/item-edit.vue

```diff
@@ -158,8 +158,12 @@ export default {
             closeTimeout: 2000
           }).open()
         }
-        this.dirty = false
-        this.$f7router.back()
+        if (this.item.metadata && this.item.metadata.unit) {
+          this.$oh.api.put('/rest/items/' + this.item.name + '/metadata/unit', this.item.metadata.unit).then((data) => {
```

Comment:
Looking at this I'm wondering why metadata is used to store the unit instead of just adding a new field to the Item.
(I'll admit I have just skimmed through the discussion at https://github.com/openhab/openhab-core/issues/3282).

Alternatively it might be interesting to have an EnrichedItemDTO accepted as request payload to `PUT /rest/items/*` and if metadata is present, sync it to the metadata registry.